### PR TITLE
Fix WarningLow/CriticalLow sensors states

### PR DESCRIPTION
**Pull request by nest1ing**
Fix the conditions for switch sensors to the WarningLow/CriticalLow
states.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

### CODE REVIEW

File: yadro/sensors.cpp [hunk 1]
```diff
@@ -70,15 +70,15 @@ sensor_t::state_t sensor_t::enable(bool s)
     {
         state = E_CRITICAL_HIGH;
     }
-    else if (criticalLow > 0 && currentValue > criticalLow)
+    else if (criticalLow > 0 && currentValue < criticalLow)
```
Comment (AlexanderAmelkin) [inline yadro/sensors.cpp]: Looks like this code assumes that the sensor value can't be negative or that the Lower Critical threshold can't be zero, which is generally not true. There may be a negative voltage sensor (e. g., -12V), or a wide range temperature sensor that can easily cross zero or have it as the lower threshold.

File: yadro/sensors.cpp [hunk 2]
```diff
@@ -70,15 +70,15 @@ sensor_t::state_t sensor_t::enable(bool s)
...
     {
         state = E_CRITICAL_LOW;
     }
     else if (warningHigh > 0 && currentValue > warningHigh)
     {
         state = E_WARNING_HIGH;
     }
-    else if (warningLow > 0 && currentValue > warningLow)
+    else if (warningLow > 0 && currentValue < warningLow)
```
Comment (AlexanderAmelkin) [inline yadro/sensors.cpp]: Same here

**Review by AlexanderAmelkin — CHANGES_REQUESTED** [review]

**Review by AlexanderAmelkin — DISMISSED** [review]

**Comment by AlexanderAmelkin** [comment]
@artemsen, any comments/objections?

**Comment by artemsen** [comment]
I don't like the architecture of this solution, but for this implementation I don't have any comments.

**Review by artemsen — APPROVED** [review]

**Pull request merged**